frontend/hub/adapters/hubErrorAdapter.tsx

```diff
@@ -0,0 +1,69 @@
+import { isRequestError } from '../../common/crud/RequestError';
+import {
+  ErrorAdapter,
+  ErrorOutput,
+  GenericError,
+} from '../../../framework/PageForm/typesErrorAdapter';
+
+export interface GalaxyError {
+  code: string;
+  detail: string;
+  source: { parameter: string };
+  status: string;
+  title: string;
+}
+
+export const hubErrorAdapter: ErrorAdapter = (error): ErrorOutput => {
+  // errors can come in several flavors depending on if the API is from
+  // pulp or ansible.
+  const genericErrors: GenericError[] = [];
+  const fieldErrors: GenericError[] = [];
+
+  if (
```

Comment:
Can there be a `ResponseError` interface that checks for these attributes rather than having this evaluated with a set of conditionals?

Reply:
Thanks for your review @kialam, I am refactoring those conditions.